### DISCUSSION TITLE
Successfully downloaded videos but cannot find them in the configed path

### DISCUSSION THREAD

**Discussion by weiwuwei** [tangyoha/telegram_media_downloader#78] [General]
I can see from logs that the files have been downloaded successfully and can see them in temp file, but cannot find them in the configed path. I am in MacOS and my configed path is `save_path: /Users/david/Download/test` (the folder do exist), anything idea? I guess there is sth wrong when moving the files

**Comment by weiwuwei** [comment]
Also, localhost:5000 doesn't work for me

**Comment by tangyoha** [comment]
> I can see from logs that the files have been downloaded successfully and can see them in temp file, but cannot find them in the configed path. I am in MacOS and my configed path is `save_path: /Users/david/Download/test` (the folder do exist), anything idea? I guess there is sth wrong when moving the files

docker？

**Reply by weiwuwei** [reply]
Yes

**Comment by tangyoha** [comment]
> Also, localhost:5000 doesn't work for me

Are you running with docker? If you are running with docker, you don’t need to modify save_path, you need to modify the path in docker-compose.yaml. Are you using vps? If it is not started locally, please add a line of web_host:0.0.0.0

**Reply by weiwuwei** [reply]
Ok, I figured out that chrome doesn't support localhost somehow, I used http://127.0.0.1:5000/ and it works. Thanks

**Comment by tangyoha** [comment]
Hello! If files are downloaded to temp but not appearing in the configured path, this is likely a permissions or path issue.

**Common Causes**:

1. **Path does not exist or lacks write permissions**
2. **Incorrect path format in config**
3. **File move operation failing silently**

**Solutions**:

**1. Verify the path exists and has correct permissions**:

```bash
# Check if directory exists
ls -la /Users/david/Download/test

# If it does not exist, create it
mkdir -p /Users/david/Download/test

# Set proper permissions
chmod 755 /Users/david/Download/test
```

**2. Check your config.yaml path format**:

```yaml
save_path: /Users/david/Download/test  # Correct
# NOT: ~/Download/test (tilde may not expand)
# NOT: /Users/david/Download/test/ (trailing slash may cause issues)
```

**3. Check logs for move errors**:

```bash
cat log/*.log | grep -i "error\|failed\|move"
```

**4. Try using absolute path without spaces**:

```yaml
save_path: /Users/david/Downloads/telegram_media  # No spaces
```

**5. Check disk space**:

```bash
df -h /Users/david/Download/test
```

**6. Manual test**:

```bash
# Try manually moving a file from temp to your save path
mv temp/test.txt /Users/david/Download/test/
```

If this fails, it confirms a permissions or path issue.

**Workaround**:

Set save_path to the temp directory temporarily to see if files stay:

```yaml
save_path: ./temp
```

Then manually move files after download completes.
